akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -150,36 +180,54 @@ final class AskableActorRef(val actorRef: ActorRef) extends AnyVal {
    * INTERNAL API: for binary compatibility
    */
   protected def ask(message: Any, timeout: Timeout): Future[Any] =
-    internalAsk(message, timeout, ActorRef.noSender)
+    internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
 
   def ask(message: Any)(implicit timeout: Timeout, sender: ActorRef = Actor.noSender): Future[Any] =
+    internalAsk(_ ⇒ message, timeout, sender)
+
+  def ask(message: ActorRef ⇒ Any)(implicit timeout: Timeout): Future[Any] =
+    internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
+
+  def ask(message: ActorRef ⇒ Any)(implicit timeout: Timeout, sender: ActorRef): Future[Any] =
     internalAsk(message, timeout, sender)
 
   /**
    * INTERNAL API: for binary compatibility
    */
   protected def ?(message: Any)(implicit timeout: Timeout): Future[Any] =
-    internalAsk(message, timeout, ActorRef.noSender)
+    internalAsk(_ ⇒ message, timeout, ActorRef.noSender)
 
   def ?(message: Any)(implicit timeout: Timeout, sender: ActorRef = Actor.noSender): Future[Any] =
+    internalAsk(_ ⇒ message, timeout, sender)
+
+  def ?(message: ActorRef ⇒ Any)(implicit timeout: Timeout): Future[Any] =
+    internalAsk(message, timeout, ActorRef.noSender)
+
+  def ?(message: ActorRef ⇒ Any)(implicit timeout: Timeout, sender: ActorRef): Future[Any] =
     internalAsk(message, timeout, sender)
```

Comment:
We should not introduce a new symbol—remember `!!` and `!!!`?—and overloading will not work (even in those cases where it does work, type inference will fail for the function literal), so providing it as another import is the only option. Thanks for the exploration!

BTW: the symmetry with `akka-typed` is the reason for doing this, explicit replyTo address style is not only useful in that context but also here—implicit `sender` is sometimes just a tad too hidden.